app/scripts/platforms/extension.js

```diff
@@ -31,6 +31,18 @@ class ExtensionPlatform {
       cb(e)
     }
   }
+
+  addMessageListener (cb) {
+    extension.runtime.onMessage.addListener(cb)
+  }
+
+  sendMessage (message, query = {}) {
+    extension.tabs.query(query, tabs => {
+      tabs.forEach(tab => {
+        extension.tabs.sendMessage(tab.id, message)
```

Comment:
Just curious,  is there any reason why you are sending the msg to every tab instead of the active one only?

Reply:
We filter the tabs two lines up based on an optional `query` parameter. This `sendMessage` method isn't meant to be MetaMask-specific, just platform-specific, so it has no notion of active or inactive tabs. It just sends a message to tabs matching a query. When we use this `sendMessage` method, we can (and usually do) pass a query of `{ active: true }` so messages only go to the active tab.

Reply:
Thanks @bitpshr. This is very useful! I’m working on the QR scanner injecting a video tag to access the camera and I didn’t want to do it on every open tab.

Reply:
Thanks @bitpshr. This is very useful! I’m working on the QR scanner injecting a video tag to access the camera and I didn’t want to do it on every open tab.